src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -207,10 +208,10 @@ private static object ValueFromScalar(ScalarGraphType scalar, object input)
         {
             if (input is IValue value)
             {
-                return scalar.ParseLiteral(value);
+                return scalar.ParseLiteral(value) ?? throw new ArgumentException($"Unable to convert '{value.Value}'");
```

Comment:
Look at `AssertValidValue`:
```c#
 if (ValueFromScalar(scalar, input) == null)
                    throw new InvalidValueException(fieldName, "Invalid Scalar value for input field.");  <-- this line is no longer needed
```
Lets inline `ValueFromScalar` method body inside `AssertValidValue` and remove `ValueFromScalar` method. And change
```c#
throw new ArgumentException($"Unable to convert '{value.Value}'");
```
to
```c#
throw new ArgumentException($"Invalid Scalar value for input field {fieldName}. Unable to convert '{value.Value}' to {type.Name}");
```

I think everyone wants to see exactly this kind of detailed error message.

Reply:
I updated it as you suggested.  This code only runs when an input variable is unable to be converted, so I have it throw an `InvalidValueException` as this inherits from `ExecutionError` and returns the field name and location information back to the caller.

Note that the location information is blocked currently due to another bug; fixed in PR #1735 which can be reviewed after this PR is merged.

Reply:
OK